joy_teleop/joy_teleop/joy_teleop.py

```diff
@@ -47,6 +46,278 @@ class JoyTeleopException(Exception):
     pass
 
 
+def get_interface_type(type_name, interface_type):
+    split = type_name.split('/')
+    if len(split) != 3:
+        raise JoyTeleopException("Invalid type_name '{}'".format(type_name))
+    package = split[0]
+    interface = split[1]
+    message = split[2]
+    if interface != interface_type:
+        raise JoyTeleopException("Cannot use interface of type '{}' for an '{}'"
+                                 .format(interface, interface_type))
+
+    mod = importlib.import_module(package + '.' + interface_type)
+    return getattr(mod, message)
+
+
+def set_member(msg, member, value):
+    ml = member.split('-')
+    if len(ml) < 1:
+        return
+    target = msg
+    for i in ml[:-1]:
+        target = getattr(target, i)
+    setattr(target, ml[-1], value)
+
+
+class JoyTeleopCommand:
+
+    def __init__(self, name, config, button_name, axes_name):
+        self.buttons = []
+        if button_name in config:
+            self.buttons = config[button_name]
+        self.axes = []
+        if axes_name in config:
+            self.axes = config[axes_name]
+
+        if len(self.buttons) == 0 and len(self.axes) == 0:
+            raise JoyTeleopException("No buttons or axes configured for command '{}'".format(name))
+
+        # This is used to short-circuit the run command if there aren't enough
+        # buttons in the message
+        self.min_button = None
+        if len(self.buttons) > 0:
+            self.min_button = min(self.buttons)
+        self.min_axis = None
+        if len(self.axes) > 0:
+            self.min_axis = min(self.axes)
+
+        # This is used to "debounce" the message; if we get multiple presses
+        # of the button(s), we only activate on the first one until it toggles
+        # again.  If there are multiple buttons, this means that mashing all of
+        # them together will only activate this topic once.
+        self.active = 0
+
+        # We use 'ready' to deal with cases where a service/action becomes
+        # ready between the last call and the current one.  That is, if we get
+        # a joystick command to call the service/action while it is not ready,
+        # followed immediately by another command to call the service/action
+        # and it has become ready, we still want to call it the second time.
+        self.ready = False
+
+    def check_buttons_and_axes(self, joy_state):
+        if (self.min_button is not None and len(joy_state.buttons) <= self.min_button) and \
+           (self.min_axis is not None and len(joy_state.axes) <= self.min_axis):
+            # There weren't enough buttons or axes in the message, so it can't possibly
+            # be a message for us.
+            return False
+
+        last_active = self.active
+
+        self.active = 0
```

Comment:
@clalancette should this be 
```suggestion
        self.active = False
```
?

Reply:
Yep, fixed in the latest.